### PR TITLE
[blazor][wasm] Remove legacy interop

### LINKED ISSUE #48042: [blazor][wasm] Remaining legacy interop
Below is list of usages of legacy interop.

- [x] Usage of `BINDING.call_assembly_entry_point`
   - https://github.com/dotnet/aspnetcore/blob/730dac21a5991eb8f130717c12de22f46715b8c7/src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts#L68
   - this one is most problematic because it uses `eval()` which is [CSP blocker](https://github.com/dotnet/aspnetcore/issues/37787)
- [ ] Usage of `BINDING.js_string_to_mono_string` in `invokeJSFromDotNet`
   - https://github.com/dotnet/aspnetcore/blob/730dac21a5991eb8f130717c12de22f46715b8c7/src/Components/Web.JS/src/Boot.WebAssembly.ts#L124
   - https://github.com/dotnet/aspnetcore/blob/730dac21a5991eb8f130717c12de22f46715b8c7/src/Components/Web.JS/src/Boot.WebAssembly.ts#L138
   - this is potentially just dead code, if the user code doesn't use the obsolete `IJSUnmarshalledRuntime`
   - but I don't fully understand `jsCallDispatcher` and `invokeJSJson` use-cases
- [ ] usage of `BINDING.unbox_mono_obj` and `BINDING.conv_string` in `readStringField` which is part of `renderBatch`
   - https://github.com/dotnet/aspnetcore/blob/730dac21a5991eb8f130717c12de22f46715b8c7/src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts#L127
   - https://github.com/dotnet/aspnetcore/blob/730dac21a5991eb8f130717c12de22f46715b8c7/src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts#L138
   - https://github.com/dotnet/aspnetcore/blob/730dac21a5991eb8f130717c12de22f46715b8c7/src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts#L142
   - this is problematic from GC safety, for which we do have stop-gap solution.
   - this is probably not a problem for CSP as there is no bind/invoke and no `eval()`
- [x] `MonoHeapLock.stringCache` is pointer hazard
   - https://github.com/dotnet/aspnetcore/blob/730dac21a5991eb8f130717c12de22f46715b8c7/src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts#L139
   - because `BINDING.conv_string` is already caching interned strings by their pointer (because they never move on GC)
   - any other strings do move and could be even collected, so the pointer could be pointing at different object and produce a) invalid value or b) crash even in single threaded app
   - this is problematic from MT GC safety, for which we do have stop-gap solution.

All of the uses prevent users trimming legacy interop dead code with `<WasmEnableLegacyJsInterop>false</WasmEnableLegacyJsInterop>`

### PR DESCRIPTION
**Pull request by pavelsavara**
- drop obsolete C# API `IJSUnmarshalledObjectReference` and `IJSUnmarshalledRuntime`
- drop `invokeJSFromDotNet()`
- drop `toUint8Array()`
- drop `readFloatField()`
- drop `getArrayLength()`
- stop using `BINDING.conv_string` and `BINDING.unbox_mono_obj`
- use `MONO_INTERNAL.monoStringToStringUnsafe` and `MONO_INTERNAL.monoObjectAsBoolOrNullUnsafe` instead

Depends on https://github.com/dotnet/runtime/pull/95869
Fixes https://github.com/dotnet/aspnetcore/issues/48042

### CODE REVIEW

File: src/Components/WebAssembly/WebAssembly/src/targets/Microsoft.AspNetCore.Components.WebAssembly.props [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <Project>
   <PropertyGroup>
     <BlazorWebAssemblyJSPath>$(MSBuildThisFileDirectory)blazor.webassembly.js</BlazorWebAssemblyJSPath>
+    <WasmEnableLegacyJsInterop Condition="'$(WasmEnableLegacyJsInterop)' == ''" >false</WasmEnableLegacyJsInterop>
```
Comment (pavelsavara) [inline src/Components/WebAssembly/WebAssembly/src/targets/Microsoft.AspNetCore.Components.WebAssembly.props:4]: @maraf is this right place to set a default for native relink with workload ?

Comment (maraf) [reply src/Components/WebAssembly/WebAssembly/src/targets/Microsoft.AspNetCore.Components.WebAssembly.props:4]: The right place in SDK https://github.com/dotnet/sdk/blob/6ad1cd1de740d3332af31bd57afef0070b7a3329/src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets#L55

Comment (pavelsavara) [reply src/Components/WebAssembly/WebAssembly/src/targets/Microsoft.AspNetCore.Components.WebAssembly.props:4]: Also https://github.com/dotnet/runtime/blob/6b6b0402953526c6b81c95c2f1cf1450fbaf036a/src/mono/wasm/build/BrowserWasmApp.targets#L68

Comment (maraf) [reply src/Components/WebAssembly/WebAssembly/src/targets/Microsoft.AspNetCore.Components.WebAssembly.props:4]: But we should set the default value on our side

**Comment by pavelsavara** [comment]
`Microsoft.AspNetCore.Components.E2ETest.Tests.InteropTest.CanInvokeDotNetMethods` fails with `e._internal.invokeJSFromDotNet is not a function`

Because it's called via `WebAssembly.JSInterop.InternalCalls::InvokeJS` (ical) -> `mono_wasm_invoke_js_blazor` (in runtime code)

**Comment by pavelsavara** [comment]
Unrelated CI failure in `IdentityUIScriptsTest.IdentityUI_ScriptTags_SubresourceIntegrityCheck`
[Log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-52732-merge-f7d9fe79091440d3aa/Microsoft.AspNetCore.Identity.Test--net9.0/1/console.fac0e20c.log?helixlogtype=result)

File: src/Components/Web.JS/src/Boot.WebAssembly.ts [hunk 2]
```diff
@@ -32,13 +32,5 @@ Blazor.start = boot;
...
-  boot().catch(error => {
-    if (typeof Module !== 'undefined' && Module.err) {
-      // Logs it, and causes the error UI to appear
-      Module.err(error);
-    } else {
-      // The error must have happened so early we didn't yet set up the error UI, so just log to console
-      console.error(error);
-    }
-  });
+  boot().catch(printErr);
```
Comment (maraf) [inline src/Components/Web.JS/src/Boot.WebAssembly.ts:35]: This change adds showing the error UI element, which seems logical IMO

**Review by maraf — APPROVED** [review]
Changes look good to me 👍

File: src/Components/Web.JS/src/Boot.WebAssembly.ts (hunk 2 above)
Comment (pavelsavara) [reply src/Components/Web.JS/src/Boot.WebAssembly.ts:35]: Original code did the same via `Module.err` which is `printErr`

**Review by javiercn — APPROVED** [review]
Looks great!

**Pull request merged**